storage/Devices/Partition.h

```diff
@@ -71,9 +71,28 @@ namespace storage
 	unsigned int get_id() const;
 	void set_id(unsigned int id);
 
-	bool get_boot() const;
+	/**
+	 * Query the boot flag of the partition.
+	 */
+	bool is_boot() const;
+
+	/**
+	 * Set the boot flag of the partition. Only supported on Msdos. Note:
+	 * The boot flag known in parted for Gpt sets the partition id to ESP
+	 * and is not a discrete flag.
```

Comment:
Partitions on GPT have no boot flag, "set <nr> boot on" with parted on GPT partitions only sets the partition type to EFI System Partition. That can also be done with "set <nr> esp on".

In general the id/flag handling in parted is very weird and I prefer to have a clear interface in the library.